### PR TITLE
[OTel] Refactor WSGI instrumentation

### PR DESCRIPTION
**Pull request by GSVarsha**
coverage: 98%

### CODE REVIEW

File: tests/frameworks/test_wsgi.py [hunk 1]
```diff
@@ -5,16 +5,17 @@
...
 
-import tests.apps.flask_app
-from ..helpers import testenv
+from tests.apps import bottle_app
+from tests.helpers import testenv
 from instana.singletons import agent, tracer
+from instana.span.span import get_current_span
 
 
 class TestWSGI(unittest.TestCase):
```
Comment (pvital) [inline tests/frameworks/test_wsgi.py:14]: Please, transform it into a pytest one removing the `unittest`.

**Review by pvital — CHANGES_REQUESTED** [review]

File: src/instana/instrumentation/wsgi.py [hunk 2]
```diff
@@ -22,38 +24,52 @@ def __call__(self, environ, start_response):
 
         def new_start_response(status, headers, exc_info=None):
```
[the remark this answers is not in the document]
Comment (GSVarsha) [reply src/instana/instrumentation/wsgi.py]: Added typehints.

File: tests/frameworks/test_wsgi.py [hunk 3]
```diff
@@ -3,383 +3,327 @@
...
+from tests.helpers import testenv
 from instana.singletons import agent, tracer
+from instana.span.span import get_current_span
 
 
-class TestWSGI(unittest.TestCase):
-    def setUp(self):
+class TestWSGI:
+    @pytest.fixture(autouse=True)
+    def _setUp(self) -> Generator[None, None, None]:
```
Comment (pvital) [inline tests/frameworks/test_wsgi.py:17]: Please, use `resource` as the method name, as requested by you :-P

**Review by pvital — CHANGES_REQUESTED** [review]

File: tests/requirements-313.txt [hunk 4]
```diff
@@ -15,6 +16,7 @@ markupsafe>=2.1.0
 # Depends on grpcio
 #google-cloud-pubsub<=2.1.0
 #google-cloud-storage>=1.24.0
+legacy-cgi>=2.6.1
```
[the remark this answers is not in the document]
Comment (GSVarsha) [reply tests/requirements-313.txt:19]: My bad, forgot to add a comment in the requirements file. Added now!

File: tests/frameworks/test_wsgi.py (hunk 3 above)
Comment (pvital) [reply tests/frameworks/test_wsgi.py:17]: Yes, please! Keeping a pattern will always be good.

**Review by pvital — APPROVED** [review]
It looks good to me.

File: tests/frameworks/test_wsgi.py (hunk 3 above)
Comment (GSVarsha) [reply tests/frameworks/test_wsgi.py]: Done!

**Review by CagriYonca — APPROVED** [review]
LGTM

**Pull request merged**